### ISSUE TITLE
Allow to load only weights overlay for MLD

### ISSUE THREAD

**Issue by oxidase** [Project-OSRM/osrm-backend#3983]
To minimize turn-around time of weight updates if the base graph partition was not modified then only [data changed by the customizer](https://github.com/Project-OSRM/osrm-backend/blob/0e39aa9488bff97cffda1d12fb18cdeb537e0c03/src/customize/customizer.cpp#L116-L125) should be updated in a shared memory block:
* [`overlay graph data `](https://github.com/Project-OSRM/osrm-backend/blob/0e39aa9488bff97cffda1d12fb18cdeb537e0c03/src/storage/storage.cpp#L906-L945)
* [MLD base graph data](https://github.com/Project-OSRM/osrm-backend/blob/0e39aa9488bff97cffda1d12fb18cdeb537e0c03/src/storage/storage.cpp#L458-L471)
* [geometries data](https://github.com/Project-OSRM/osrm-backend/blob/0e39aa9488bff97cffda1d12fb18cdeb537e0c03/src/updater/updater.cpp#L510)
* [datasource](https://github.com/Project-OSRM/osrm-backend/blob/0e39aa9488bff97cffda1d12fb18cdeb537e0c03/src/updater/updater.cpp#L370)
* [turn penalties](https://github.com/Project-OSRM/osrm-backend/blob/0e39aa9488bff97cffda1d12fb18cdeb537e0c03/src/updater/updater.cpp#L670-L671)

Steps to implement:
- [ ] Add a command line option to `osrm-datastore`
- [ ] Add block sizes checks without [requesting memory](https://github.com/Project-OSRM/osrm-backend/blob/0e39aa9488bff97cffda1d12fb18cdeb537e0c03/src/storage/storage.cpp#L431-L443). `PopulateLayout` must throw an exception if data sizes in file mismatch allocated blocks sizes. Here can be useful a check of the loaded base graph and overlay data fingerprints to detect data inconsistency, but it will require  more general solution.
- [ ] Implement conditional data loading.
- [ ] Unit tests to cover partial overlay updates (successful and failing paths)

**Comment by TheMarex** [comment]
Good idea. 👍 

From an API standpoint I think ti would make sense to implement this by making `osrm-datastore` not only accept a `.osrm` path prefix, but also accept a list of files:

```
osrm-datastore data.osrm.mldgr data.osrm.cells data.osrm.geometries
```

I would like to keep this on a per-file basis, because I think having the abstraction of one file == one coherent piece of data makes sense.

With multi-metric support we will need to split some of these files anyway to only duplicated the weights portion of every data structure.

**Comment by TheMarex** [comment]
This depends on https://github.com/Project-OSRM/osrm-backend/issues/4873 but is not a blocker for multi-metric.

**Comment by TheMarex** [comment]
After #4975 is merged and #4873 fixed I would propose the following change to how `osrm-datastore` works:

#### Full data update (current behavior)
```
./osrm-datstore data.osrm
```
Will allocate two new shared memory regions, one for the base dataset and one for the metric and load data into them.

#### Only metric update
```
./osrm-datastore --only-metrics data.osrm
```

Will only allocate a new shared memory region for the metric and update it. What a "metric" is depends on the algorithm, but for CH that is basically the whole contracted graph, for MLD that is (for now) the base graph and all overlay data.

This could be extensible for multi-metric in the future, were we only want to update certain metrics:

```
./osrm-datastore --only-metrics=traffic_durations data.osrm
```
